src/domains/printable.ml

```diff
@@ -87,13 +87,14 @@ struct
   let pretty_diff () (x,y) =
     dprintf "%s: %a not leq %a" (P.name ()) pretty x pretty y
   let printXml f x = BatPrintf.fprintf f "<value>\n<data>\n%s\n</data>\n</value>\n" (XmlUtil.escape (P.show x))
+  let to_yojson x = `String (P.show x)
 end
 
 
 module type Name = sig val name: string end
 module UnitConf (N: Name) =
 struct
-  type t = unit [@@deriving eq, ord, yojson]
+  type t = unit [@to_yojson fun () -> `String N.name] [@@deriving eq, ord, to_yojson]
```

Comment:
Better just make this a normal `let to_yojson` because it practically is manually defining it anyway.